File: C#/0069. Sqrt(x).cs
public class Solution {
    public int MySqrt(int x) {
        if(x==1){
            return 1;
        }
        long left=0;
        long right=x/2;
        long mid;
        long result=0;
        while(left<=right){
            mid=(left+right)/2;

            if(mid*mid==x){
                result=mid;
                break;
            }
            if(mid*mid>int.MaxValue){
                right=mid-1;
            }
            else if(mid*mid>x){
                right=mid-1;
            }
            else{
                left=mid+1;
            }
        }
        
        if(result==0){
            result=right;
        }
        return (int)result;
    }
}

File: 20190519python/0069. Sqrt(x).py
class Solution:
    def mySqrt(self, x: int) -> int:
        lo=1
        hi=x
        while lo<=hi:
            mid=lo+(hi-lo)//2
            if mid**2==x:
                return mid
            elif mid**2<x:
                lo=mid+1
            else:
                hi=mid-1
        return lo-1
